### PR TITLE
Make code TypeScript-compatible again so runtime errors can be minimized

### PR DESCRIPTION
**Pull request by heartdisease**
- Re-introduced a tsconfig.json and a Webpack configuration for building and minifying the script + CSS
- Introduced .editorconfig and Prettier for consistent formatting across IDEs
- Added entries to .gitignore to also exclude local IDE project files and node_modules
- Added .gitattributes to ensure consistent line endings in all text files (LF)

### CODE REVIEW

File: bookmarklet/bookmarklet.css [hunk 1]
```diff
@@ -1,592 +1,612 @@
...
+  opacity: 0;
 }
 
 .pure-material-button-text:disabled::after {
-	opacity: 0;
+  opacity: 0;
 }
 
-/** filament group: Styling a Select Like It�s 2019 ***********************************************************/
+/** filament group: Styling a Select Like It�s 2019 ***********************************************************/
```
Comment (christian-fischer) [inline bookmarklet/bookmarklet.css:397]: looks like something went wrong here?

File: bookmarklet/bookmarklet.ts [hunk 2]
```diff
@@ -3,576 +3,688 @@ import './bookmarklet.css';
...
+        const idFooterRight =
+          'com_kamilsarelo_dynatrace_timetracking_footer_right';
+        const idSelectType =
+          'com_kamilsarelo_dynatrace_timetracking_select_type';
+        const idButtonBook =
+          'com_kamilsarelo_dynatrace_timetracking_action_book';
+        const idButtonClear =
+          'com_kamilsarelo_dynatrace_timetracking_action_clear';
+        const idButtonClose =
+          'com_kamilsarelo_dynatrace_timetracking_action_close';
```
Comment (christian-fischer) [inline bookmarklet/bookmarklet.ts:111]: this looks weird; tbh. line length 80 is so last century.
I propose to increase this to something more sensible like maybe 120

Comment (heartdisease) [reply bookmarklet/bookmarklet.ts:111]: I'd strongly advise again this! 80 cols is the Prettier default and they state explicitly on their website that it should not be altered (especially not to 120, 100 is already too much for the way Prettier formats code). Also TypeScript is not as verbose as Java is and we only use 2 spaces as indentation here. Vertical code is more readable to the human eye anyway. May be that in this particular case it looks rather ... awkward. But if the code would be more well-structured, the indentation would be much shorter here and Prettier wouldn't wrap it.

File: bookmarklet/bookmarklet.css (hunk 1 above)
Comment (heartdisease) [reply bookmarklet/bookmarklet.css:397]: Maybe the ' here was a weird Unicode character... I'll have a look.

**Pull request merged**